src/ucp/rndv/proto_rndv.c

```diff
@@ -715,19 +715,12 @@ ucp_proto_rndv_ppln_frag_sys_dev(ucp_context_h context,
                                  ucs_memory_type_t mem_type,
                                  ucs_sys_device_t sys_dev)
 {
-    ucs_status_t status;
-    ucp_md_index_t md_index;
-    ucp_memory_info_t mem_info;
-
     if (sys_dev != UCS_SYS_DEVICE_ID_UNKNOWN) {
         return sys_dev;
     }
 
-    status = ucp_mm_get_alloc_md_index(context, mem_type,
-                                       UCS_SYS_DEVICE_ID_UNKNOWN, &md_index,
-                                       &mem_info);
-    if ((status == UCS_OK) && (md_index != UCP_NULL_RESOURCE)) {
-        return mem_info.sys_dev;
+    if (context->alloc_md[mem_type].initialized) {
+        return context->alloc_md[mem_type].sys_dev;
```

Comment:
just return `context->alloc_md[mem_type].sys_dev`, no need to check for initialized